core/controllers/access_validators_test.py

```diff
@@ -615,6 +619,74 @@ def test_validation_returns_false_if_given_user_is_non_existent(
         self.logout()
 
 
+class SkillEditorPageAccessValidationHandlerTests(test_utils.EmailTestBase):
+    """Checks the access to the skill editor page and its rendenring"""
+
+    def setUp(self) -> None:
+        super().setUp()
+        self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
+        self.add_user_role(
+            self.CURRICULUM_ADMIN_USERNAME, feconf.ROLE_ID_CURRICULUM_ADMIN)
+
+        self.admin_id = self.get_user_id_from_email(self.CURRICULUM_ADMIN_EMAIL)
+
+        self.skill_id = skill_services.get_new_skill_id()
+        self.save_new_skill(
+            self.skill_id, self.admin_id, description='Skill Description')
+        self.skill_id_2 = skill_services.get_new_skill_id()
+        self.save_new_skill(
+            self.skill_id_2, self.admin_id, description='Skill Description 2')
+
+    def test_access_skill_editor_page_without_logging_in(self) -> None:
+        self.get_json(
+            '%s/can_access_skill_editor/%s' % (
+            ACCESS_VALIDATION_HANDLER_PREFIX, self.skill_id
+            ), expected_status_int=401
+        )
+
+    def test_access_skill_editor_page_with_guest_user(self) -> None:
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+        self.login(self.VIEWER_EMAIL)
+        self.get_json(
+            '%s/can_access_skill_editor/%s' % (
+            ACCESS_VALIDATION_HANDLER_PREFIX, self.skill_id
+            ), expected_status_int=401
+        )
+        self.logout()
+
+        self.login(self.NEW_USER_EMAIL)
+        self.get_json(
+            '%s/can_access_skill_editor/%s' % (
+            ACCESS_VALIDATION_HANDLER_PREFIX, self.skill_id
+            ), expected_status_int=401
+        )
+        self.logout()
```

Comment:
What's the difference between a viewer and a new user? Because either:
- There are discernible differences between both user roles, so they should be in their own separate test OR
- There is no difference between both user roles, so we should drop one of these group of statements to avoid exercising duplicate behaviors